Formula/pocl.rb

```diff
@@ -0,0 +1,51 @@
+class Pocl < Formula
+  desc "Portable Computing Language"
+  homepage "http://portablecl.org"
+  url "http://portablecl.org/downloads/pocl-3.0.tar.gz"
+  sha256 "a3fd3889ef7854b90b8e4c7899c5de48b7494bf770e39fba5ad268a5cbcc719d"
+  license "MIT"
+  head "https://github.com/pocl/pocl.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "opencl-headers" => :build
+  depends_on "pkg-config" => :build
+  depends_on "hwloc"
+  depends_on "llvm"
+  depends_on "ocl-icd"
+
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5" # LLVM is built with GCC
+
+  def install
+    args = %W[
+      -DCMAKE_INSTALL_RPATH=#{lib};#{lib}/pocl
+      -DENABLE_EXAMPLES=OFF
+      -DENABLE_TESTS=OFF
+      -DLLVM_BINDIR=#{Formula["llvm"].opt_bin}
+      -DPOCL_INSTALL_ICD_VENDORDIR=#{prefix}/etc/OpenCL/vendors
+    ]
+    # Avoid installing a copy of OpenCL headers on macOS
+    args << "-DOPENCL_H=#{Formula["opencl-headers"].opt_include}/CL/opencl.h" if OS.mac?
+    # Only x86_64 supports "distro" which allows runtime detection of SSE/AVX
+    args << "-DKERNELLIB_HOST_CPU_VARIANTS=distro" if Hardware::CPU.intel?
+
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    pkgshare.install "examples"
+    # We use a symlink for the ICD file as it contains the real Cellar path to the
+    # shared library and needs to be kept up-to-date to work with the ICD loader.
+    (etc/"OpenCL/vendors").install_symlink prefix/"etc/OpenCL/vendors/pocl.icd"
```

Comment:
Brew DSL doesn't handle `etc` well in this scenario as seen by various issues I mentioned in previous comment. Particularly, `etc.install_symlink` isn't well defined and behaves inconsistently.

I am now considering either manually creating ICD file or manually creating opt path symlink. 

```suggestion
    # The default ICD file contains the realpath to shared library, which breaks if
    # not overwritten. So, we cannot directly install generated ICD file into etc.
    # Symlinking files into etc exhibit unreliable behavior as it currently
    # overwrites files, which doesn't align with how etc.install behaves. Also,
    # manually running etc.install_symlink can produce broken symlinks on updates.
    # So, we manually create ICD file to fixed opt path with unversioned library.
    (buildpath/"pocl.icd").write opt_lib/shared_library("libpocl")
    (etc/"OpenCL/vendors").install "pocl.icd"
```

```suggestion
    # The default ICD file contains the realpath to shared library, which breaks if
    # not overwritten. So, we cannot directly install generated ICD file into etc.
    # etc.install_symlink into exhibits unreliable behavior as it currently
    # overwrites files, which doesn't align with how etc.install behaves. Also,
    # manually running etc.install_symlink can produce broken symlinks on updates.
    # So, we manually create symlink to ICD file if it doesn't exist.
    (etc/"OpenCL/vendors").mkpath
    icd = etc/"OpenCL/vendors/pocl.icd"
    ln_s opt_libexec/"OpenCL/vendors/pocl.icd", icd unless icd.exist?
```